### ISSUE TITLE
Bug in a CategorialImputer test.

### ISSUE THREAD

**Issue by arnau126** [paulgb/sklearn-pandas#86]
CategorialImputer tests are not actually testing the feature when `input_type is 'np'` and `none_value is np.nan`.

This is because of the use of `np.asarray`. This function infers the data type and in this case it is casting the np.nan to string.

In explanation:
```
In [1]:  data = ['a', 'b', 'b', np.nan]

In [2]: X = np.asarray(data)

In [3]: X
Out [3]: array(['a', 'b', 'b', 'nan'], 
            dtype='|S3')

```

So, in this case, there is no null-like value and the imputer is doing nothing.

**Comment by dukebody** [comment]
Thanks @arnau126 ! Can you submit a PR to fix this?

**Comment by dukebody** [comment]
If you have any other improvements to the `CategoricalImputer` in mind (you probably do) please chime in on that as well.

**Comment by arnau126** [comment]
Yes, I do :)
PR: #87

**Comment by dukebody** [comment]
Fixed in #87.
